### ISSUE TITLE
Using Redux-Forms Field, onValueChange doesn't no show onBlur event

### ISSUE THREAD

**Issue by wontwon** [s-yadav/react-number-format#199]
Hello, just having an issue currently with Redux Forms and your package. Currently, when using onValueChange, I can see that the event shows "change" when typing into an input. However, when clicking out of the input, the event doesn't reflect "blur". 

Any thoughts?

**Comment by s-yadav** [comment]
onValueChange gets called whenever value changes, either in onChange event or onBlur. If there is no change on input value it will not get called. You might want to add onBlur prop to listen that event.

**Comment by wontwon** [comment]
@s-yadav thanks for the response. I'm not quite sure I follow. According to the docs, onValueChange should have an event of onChange or onBlur. I'm implementing it like so: 

```
<NumberFormat
          id={id}
          decimalScale={4}
          name={name}
          allowNegative={false}
          displayType={type || "text"}
          className={className || "form-control"}
          value={input.value || value || ""}
          onValueChange={setPriceValue}
          tabIndex={tabIndex}
          disabled={disabled}
        />
```

you are suggesting:

```
<NumberFormat
          id={id}
          decimalScale={4}
          name={name}
          allowNegative={false}
          displayType={type || "text"}
          className={className || "form-control"}
          value={input.value || value || ""}
          onValueChange={setPriceValue}
          onBlur={setValueOnBlur}
          tabIndex={tabIndex}
          disabled={disabled}
        />
```

**Comment by wontwon** [comment]
@s-yadav the onValueChange description: onValueChange handler accepts values object and event object (change / blur) as second parameter.

When I change, the input value. I see this when I log out the event:
![devtools_-_localhost_9000_replenishment_purchase-order_express_and_zoom](https://user-images.githubusercontent.com/3239936/42513509-1d33c04c-8425-11e8-872d-369fa7ef8646.png)

But when I blur the the input by clicking out. Nothing gets logged. Am I misunderstanding the documentation?

**Comment by s-yadav** [comment]
Yes, so as mentioned in previous comment onValueChange is called when ever there is a change on formatted value. Which mostly happens on onChange, but there are some transformation component applies on onBlur event in which case onValueChange is called. It does not guarantee that in all onBlur onValueChange will be called. But whenever called you will get second param as onBlur event. 

May be I should improve the doc to convey this properly. 

And what I am suggesting is 
```jsx
<NumberFormat
          id={id}
          decimalScale={4}
          name={name}
          allowNegative={false}
          displayType={type || "text"}
          className={className || "form-control"}
          value={input.value || value || ""}
          onValueChange={setPriceValue}
          onBlur={doSomethingOnBlur}
          tabIndex={tabIndex}
          disabled={disabled}
        />
```
doSomethingOnBlur can get value from state which you might be setting on setPriceValue

**Comment by s-yadav** [comment]
onValueChange should not receive event as second argument. This cause confusion. Also required for fix of #227. Will be removed on v4.

**Comment by EgorEres** [comment]
Why you do it? Any reason to remove onBlur, it's weird, I don't want to execute my logic on every change, I want to do it one time when I lose focus.
